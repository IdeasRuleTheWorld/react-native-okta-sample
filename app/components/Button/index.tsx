import React from 'react';
import { Text, View, Button } from 'native-base';
import { StyleSheet } from 'react-native';
import colors from '../../config/colors';
import fonts from '../../config/fonts';

const ButtonClass = props => {
  const { onPress, btnstyle, btnTitle } = props;
  return (
    <View style={{ margin: 5 }}>
      <Button
        onPress={onPress}
        style={btnstyle === 'solid' ? buttonStyles.solidButton : buttonStyles.borderedButton}
      >
        <Text uppercase={false}>{btnTitle}</Text>
      </Button>
    </View>
  );
};

export default ButtonClass;

const buttonStyles = StyleSheet.create({
  solidButton: {
    backgroundColor: colors.$DFS_red,
    borderRadius: 3,
    borderColor: colors.$DFS_red,
    borderWidth: 1,
    marginLeft: 20,
    marginRight: 20,
    marginTop: 10,
    justifyContent: 'center',
    height: 40,
    color: colors.$white3,
    fontSize: 18,
    fontWeight: '500',
    fontFamily: fonts.$fontFamilyMedium,
    paddingVertical: 5,
  },
  borderedButton: {
    backgroundColor: colors.$white3,
    borderRadius: 3,
    borderColor: colors.$DFS_red,
    borderWidth: 1,
    marginLeft: 20,
    marginRight: 20,
    marginTop: 10,
    justifyContent: 'center',
    height: 40,
    color: colors.$DFS_red,
    fontSize: 18,
    fontWeight: '500',
    fontFamily: fonts.$fontFamilyMedium,
    paddingVertical: 5,
  },
});
