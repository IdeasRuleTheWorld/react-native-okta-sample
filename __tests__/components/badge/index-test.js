import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { BadgeClass } from "../../../app/components/badge/";

describe('Badge suite', () => {
    it("Render badge with active=true", () => {
        const props = {
            active: true,
            textContent: 'Sample Text',
            size: 10,
            fontSize: 10,
        }
        const root = renderer.create(<BadgeClass {...props} />).toJSON();
        expect(root).toMatchSnapshot();
    });
    it("Render badge with active=false", () => {
        const props = {
            active: false,
            textContent: 'Sample Text',
            size: 10,
            fontSize: 10,
        }
        const root = renderer.create(<BadgeClass {...props} />).toJSON();
        expect(root).toMatchSnapshot();
    });
});

/** before each, after each, before all, after all */
