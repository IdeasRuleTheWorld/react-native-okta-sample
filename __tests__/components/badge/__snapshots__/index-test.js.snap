// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge suite Render badge with active=false 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#ED1727",
        "borderRadius": 13.5,
        "height": 27,
        "justifyContent": "center",
        "paddingBottom": 3,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 3,
      },
      Object {
        "backgroundColor": "#c0311a",
      },
      "",
      Object {
        "height": 10,
        "minWidth": 10,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 15,
          "lineHeight": 19,
          "paddingLeft": 3,
          "paddingRight": 3,
          "textAlign": "center",
        },
        Object {
          "color": "#ffffff",
        },
        "",
        Object {
          "fontSize": 10,
        },
        Object {
          "lineHeight": 10,
        },
      ]
    }
    uppercase={false}
  >
    Sample Text
  </Text>
</View>
`;

exports[`Badge suite Render badge with active=true 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#ED1727",
        "borderRadius": 13.5,
        "height": 27,
        "justifyContent": "center",
        "paddingBottom": 3,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 3,
      },
      Object {
        "backgroundColor": "#c0311a",
      },
      Object {
        "backgroundColor": "#ffffff",
      },
      Object {
        "height": 10,
        "minWidth": 10,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 15,
          "lineHeight": 19,
          "paddingLeft": 3,
          "paddingRight": 3,
          "textAlign": "center",
        },
        Object {
          "color": "#ffffff",
        },
        Object {
          "color": "#c0311a",
        },
        Object {
          "fontSize": 10,
        },
        Object {
          "lineHeight": 10,
        },
      ]
    }
    uppercase={false}
  >
    Sample Text
  </Text>
</View>
`;
